agreement_dynamic/models/agreement_section.py

```diff
@@ -44,16 +86,74 @@ def onchange_copyvalue(self):
             self.copyvalue = "${{object.{}.{} or {}}}".format(
                 self.field_id.name,
                 self.sub_model_object_field_id.name,
-                self.default_value or "''",
+                self._get_proper_default_value(),
             )
 
     # compute the dynamic content for jinja expression
     def _compute_dynamic_content(self):
-        MailTemplates = self.env["mail.template"]
         for this in self:
-            content = MailTemplates._render_template(
-                this.content, "agreement.section", [this.id]
-            )[this.id]
+            try:
+                content = this._render_template(
+                    this.content,
+                    this.resource_ref_model_id.model,
+                    this.res_id,
+                    datas={"h1": Header(), "h2": Header(), "h3": Header()},
```

Comment:
Actually if you define them here, they get reset with the rendering of each section; so they need to be defined on rendering of the whole agreement and then passed down to here as variables.